### PR TITLE
Implementation of Horizontalstepper for event creation

### LINKED ISSUE #978: Creating a Stepped approach for Event Creation
### Feature Request

**Feature Description**
Currently the Create Event is done only in a single screen in a Fragment. It would be more helpful for the user if it is done in a stepped manner. Thereby I propose the implementation of a horizontal stepper for creating an event.

**Screenshots**
tabs

**Would you like to work on the issue?**
Yes

@iamareebjamal I would like to work on adding this feature.

### PR DESCRIPTION
**Pull request by sridharjajoo**
Fixes #978 

 Screenshots for the change: 
ezgif com-resize 28

@iamareebjamal I have done a lot of changes and events are now getting created as well. I have updated the UI. Can you suggest any more changes to it?

### CODE REVIEW

**Comment by iamareebjamal** [comment]
@sridharjajoo First ask only the required information and the optional stuff should be skippable

**Comment by sridharjajoo** [comment]
@iamareebjamal In the first screen only the required information i.e event name and the duration are asked.

**Comment by sridharjajoo** [comment]
@Masquerade0097 @srv-twry @mishuvs If possible can you test this on your device and suggest any changes to it.

**Comment by iamareebjamal** [comment]
Please resolve conflicts

**Comment by iamareebjamal** [comment]
@srv-twry @Masquerade0097 @mishuvs Please review

File: app/src/main/java/org/fossasia/openevent/app/core/event/create/EventDetailsStepThree.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    }
+
+    @Override
+    public void onSuccess(String message) {
+        Toast.makeText(getActivity(), message, Toast.LENGTH_SHORT).show();
+    }
+
+    @Override
+    public void close(boolean bool) {
+        getActivity().finish();
```
Comment (Masquerade0097) [inline app/src/main/java/org/fossasia/openevent/app/core/event/create/EventDetailsStepThree.java]: The parameter `bool` is not used.

File: app/src/main/java/org/fossasia/openevent/app/core/event/create/UpdateEventFragment.java [hunk 2]
```diff
@@ -278,7 +268,7 @@ public void onSuccess(String message) {
     }
 
     @Override
-    public void close() {
+    public void close(boolean bool) {
         getActivity().finish();
```
Comment (Masquerade0097) [inline app/src/main/java/org/fossasia/openevent/app/core/event/create/UpdateEventFragment.java]: Same here.

File: app/src/main/res/layout/create_event_activity.xml [hunk 3]
```diff
@@ -6,6 +6,69 @@
...
+            android:layout_alignParentStart="true"
+            android:layout_alignParentTop="true"
+            android:layout_marginTop="100dp" />
+
+        <com.badoualy.stepperindicator.StepperIndicator
+            android:id="@+id/stepper_indicator"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginLeft="16dp"
+            android:layout_marginRight="16dp"
```
Comment (Masquerade0097) [inline app/src/main/res/layout/create_event_activity.xml]: Use dimens resource. `spacing_normal`.

File: app/src/main/res/layout/create_event_activity.xml [hunk 4]
```diff
@@ -6,6 +6,69 @@
...
+            android:orientation="horizontal"
+            android:background="@color/color_accent">
+
+            <Button
+                android:id="@+id/btn_prev"
+                android:layout_width="0dp"
+                android:layout_weight="0.5"
+                android:layout_height="wrap_content"
+                android:background="@color/color_accent"
+                android:text="Previous"
```
Comment (Masquerade0097) [inline app/src/main/res/layout/create_event_activity.xml]: Use String resource for text

File: app/src/main/res/layout/create_event_activity.xml [hunk 5]
```diff
@@ -6,6 +6,69 @@
...
+                android:layout_gravity="bottom|start"
+                android:textColor="@android:color/white" />
+
+            <Button
+                android:id="@+id/btn_next"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:background="@color/color_accent"
+                android:text="Next"
```
Comment (Masquerade0097) [inline app/src/main/res/layout/create_event_activity.xml]: Same here.

File: app/src/main/res/layout/create_event_activity.xml [hunk 6]
```diff
@@ -6,6 +6,69 @@
...
+                android:layout_weight="0.5"
+                android:layout_gravity="bottom"
+                android:textColor="@android:color/white" />
+
+            <Button
+                android:id="@+id/btn_submit"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:background="@color/color_accent"
+                android:text="Create"
```
Comment (Masquerade0097) [inline app/src/main/res/layout/create_event_activity.xml]: Same here.

File: app/src/main/java/org/fossasia/openevent/app/core/event/create/EventDetailsStepOne.java [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import timber.log.Timber;
+
+import static android.app.Activity.RESULT_OK;
+
+
```
Comment (Masquerade0097) [inline app/src/main/java/org/fossasia/openevent/app/core/event/create/EventDetailsStepOne.java]: Remove Blank line.

**Comment by Masquerade0097** [comment]
@sridharjajoo When the create event step one screen is opened, two buttons are shown in wich the left button is named `Next` and the other one with no text on it. When user clicks on any field in the form, it changes to one button only with text `Next` in center.

Also there are many PMD and Checkstyle violations, please resolve them too.

**Comment by sridharjajoo** [comment]
@iamareebjamal Should I remove the fields as mentioned above?

File: app/src/main/java/org/fossasia/openevent/app/core/event/create/CreateEventViewModel.java [hunk 8]
```diff
@@ -56,39 +67,51 @@ public CreateEventPresenter(EventRepository eventRepository, CurrencyUtils curre
...
     }
 
+    public LiveData<Event> getEventLiveData() {
+        eventMutableLiveData.setValue(event);
+        return eventMutableLiveData;
+    }
+
     private boolean verify() {
+        if (event.getName() == null) {
+            onError.setValue("Event Name cannot be empty");
```
Comment (Masquerade0097) [inline app/src/main/java/org/fossasia/openevent/app/core/event/create/CreateEventViewModel.java]: This check is already done in layout file by `validateEmpty=true`. So, this shouldn't be needed.

File: app/src/main/java/org/fossasia/openevent/app/core/event/create/PagerAdapter.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+            case 2:
+                return EventDetailsStepThree.newInstance();
+            default:
+                return null;
+        }
+    }
+
+    @Override
+    public int getCount() {
+        return 3;
```
Comment (Masquerade0097) [inline app/src/main/java/org/fossasia/openevent/app/core/event/create/PagerAdapter.java]: Please replace it with constant variable.

File: app/src/main/res/layout/create_event_activity.xml [hunk 10]
```diff
@@ -6,6 +6,72 @@
...
+            android:layout_alignParentTop="true"
+            android:layout_marginTop="100dp" />
+
+        <com.badoualy.stepperindicator.StepperIndicator
+            android:id="@+id/stepper_indicator"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginLeft="@dimen/spacing_normal"
+            android:layout_marginRight="@dimen/spacing_normal"
+            android:layout_marginTop="32dp"
```
Comment (Masquerade0097) [inline app/src/main/res/layout/create_event_activity.xml]: Use resource `spacing_large`

**Comment by sridharjajoo** [comment]
@Masquerade0097 I have done the changes. The issue you were talking about on the step one , I am not able to reproduce that.

**Comment by sridharjajoo** [comment]
@Masquerade0097 Please review and approve if no more changes are required.

**Comment by Masquerade0097** [comment]
This is the gif for the error. 
Device Details - Moto G4 Plus, Nougat


20180707185010-7c011ad592 gif-2-mp4 com

**Comment by iamareebjamal** [comment]
Create an issue for double elevation in tab and action bar please

**Comment by sridharjajoo** [comment]
@Masquerade0097 Done the changes.

**Comment by iamareebjamal** [comment]
@sridharjajoo I have merged as this was long standing PR. Please follow @mishuvs's suggestion in a follow up PR

**Pull request merged**